frontend/lib/src/components/shared/WindowDimensions/useWindowDimensions.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback, useEffect, useState } from "react"
+
+import { useTheme } from "@emotion/react"
+
+import { WindowDimensions } from "."
+
+const convertScssRemValueToPixels = (scssValue: string): number => {
+  const remValue = parseFloat(scssValue)
+  return (
+    remValue * parseFloat(getComputedStyle(document.documentElement).fontSize)
+  )
+}
+
+export const useWindowDimensions = (): WindowDimensions => {
+  const theme = useTheme()
+  const [windowDimensions, setWindowDimensions] = useState<WindowDimensions>({
+    fullWidth: 0,
+    fullHeight: 0,
+  })
+
+  const getWindowDimensions = useCallback((): WindowDimensions => {
+    const padding = convertScssRemValueToPixels(theme.spacing.md)
+    const paddingTop = convertScssRemValueToPixels(
+      theme.sizes.fullScreenHeaderHeight
+    )
+
+    return {
+      fullWidth: window.innerWidth - padding * 2, // Left and right
+      fullHeight: window.innerHeight - (padding + paddingTop), // Bottom and Top
```

Comment:
nit: if `padding` represents `Bottom`, I think we should name the variable `paddingBottom`

Reply:
It's named padding because its the padding that's on all sides. paddingTop specifically is called out for the header